src/components/DeviceTile.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from "react";
+import { LocalTrack } from "../containers/Client";
+import { useStore } from "../containers/RoomsContext";
+import { DeviceInfo } from "../containers/StreamingSettingsCard";
+import AudioVisualizer from "./AudioVisualizer";
+import { CloseButton } from "./CloseButton";
+import { DeviceIdToStream, StreamInfo } from "./StreamingDeviceSelector";
+import VideoPlayer from "./VideoPlayer";
+
+type Props = {
+  selectedId: DeviceInfo | null;
+  setSelectedId: (device: DeviceInfo | null) => void;
+  streamInfo: StreamInfo;
+  id: string;
+  streamedTracks: LocalTrack[];
+  addAudioTrack: (track: MediaStream) => void;
+  setActiveStreams: (setter: ((prev: DeviceIdToStream | null) => DeviceIdToStream) | DeviceIdToStream | null) => void;
+  addVideoTrack: (track: MediaStream) => void;
+  activeStreams: DeviceIdToStream | null;
+};
+
+const getDeviceType = (stream: MediaStream) => {
+  if (stream.getVideoTracks().length > 0) {
+    return "video";
+  } else {
+    return "audio";
+  }
+};
+
+export const DeviceTile = ({ selectedId, setSelectedId, streamInfo, id, setActiveStreams, streamedTracks }: Props) => {
+  const { state, dispatch } = useStore();
+  const isVideo = streamInfo.stream.getVideoTracks().length > 0;
+  const [enabled, setEnabled] = useState<boolean>(true);
```

Comment:
You should store streams in some kind of context because everything is lost when you switch the room tab.